File: Books-Store-FrontEnd/src/app/model/purchase.ts

import { BillingAddress } from "./billing-address";
import { CartItem } from "./cart-item";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";
import { ShippingAddress } from "./shipping-address";

export class Purchase {
    customer: Customer;
    shippingAddress: ShippingAddress;
    billingAddress: BillingAddress;
    order: Order;
    orderItems: OrderItem[];
}

File: BooksStrore-BackEnd/src/main/java/com/spring/BooksStrore/dto/Purchase.java
package com.spring.BooksStrore.dto;

import com.spring.BooksStrore.entity.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Orders order;
    private Set<OrderItem> orderItems;
}
